### PR TITLE
Write job to upload scores to bCourses

### PR DESCRIPTION
**Pull request by knrafto**
Example output: http://ok-prod.cs61a.org/admin/course/4/jobs/76/

### CODE REVIEW

File: server/canvas/api.py [hunk 1]
```diff
@@ -12,14 +12,22 @@ def canvas_api_request(canvas_course, method, endpoint, **kwargs):
...
+    }
+    params = kwargs.pop('params', {})
+    params['per_page'] = 100
+    response = requests.get(url, headers=headers, params=params, **kwargs)
+    response.raise_for_status()
+    results = response.json()
+    while 'next' in response.links:
+        response = requests.get(response.links['next']['url'], headers=headers)
+        response.raise_for_status()
+        results.extend(response.json())
```
Comment (Sumukh) [inline server/canvas/api.py]: Have you ever hit the API rate limit while doing this? 

We haven't hit the rate limit in the past - so possibly not worth worrying about - but if we do have to - we can use the response headers to figure out how many requests we have left.

File: server/canvas/api.py [hunk 2]
```diff
@@ -30,15 +38,31 @@ def get_course(canvas_course):
...
 def get_assignments(canvas_course):
-    return canvas_api_request(
+    return canvas_api_get_list(
         canvas_course,
-        'GET',
-        '/courses/{}/assignments?per_page=100'.format(canvas_course.external_id),
+        '/courses/{}/assignments'.format(canvas_course.external_id),
+    )
+
+def get_students(canvas_course):
```
Comment (Sumukh) [inline server/canvas/api.py]: Sidenote: Could this enable automatic/recurring enrolling of students?

File: server/canvas/upload.py [hunk 3]
```diff
@@ -1,16 +1,64 @@
...
+        else:
+            max_score = max(scores, key=lambda score: score.score)
+            new_score = max_score.score
+            backup_id = encode_id(max_score.backup_id)
+            if old_score != new_score:
+                new_scores[canvas_user_id] = new_score
+                stats['updated'] += 1
+            else:
+                stats['not_changed'] += 1
+        logger.info(row_format.format(sid, emails, backup_id, old_score, new_score))
```
Comment (Sumukh) [inline server/canvas/upload.py]: If it's a group - is it ok to just print one SID? 

Also - why we need to print SIDs? It's somewhat sensitive information - so the less we display it the better. If it makes it much easier to debug in bCourses - then we can keep it

File: server/canvas/api.py (hunk 1 above)
Comment (cycomachead) [inline server/canvas/api.py]: It heavily depends on the endpoint. If you're just iterating over students or assignments you should almost never the limit.

File: server/canvas/upload.py (hunk 3 above)
Comment (knrafto) [inline server/canvas/upload.py]: (Re: groups) We're iterating over students, not groups. So if we have a group of two students, both students will be printed separately and both students will get a score (it should be the same score, though).

File: server/canvas/api.py (hunk 1 above)
Comment (knrafto) [inline server/canvas/api.py]: We can cache getting all students for a bit (since every upload has to do this).

**Pull request merged**